lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -748,15 +765,24 @@ def _set_node_conf(self, nodes: List[Node]) -> None:
                 # indicates that this unit is part of the "initial cm nodes"
                 self.peers_data.put(Scope.UNIT, "bootstrap_contributor", True)
 
-        self.opensearch_config.set_node(
-            self.app.name,
-            self.model.name,
-            self.unit_name,
-            computed_roles,
-            cm_names,
-            cm_ips,
-            contribute_to_bootstrap,
-        )
+        # Update config only when we have meaningful changes.
+        current_config = self.opensearch_config.load_node()
```

Comment:
This logic should moved to `_reconfigure_and_restart_unit_if_needed` (refer to previous comment)